text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+1. It is neither a valid [tag name](https://html.spec.whatwg.org/multipage/syntax.html#syntax-tag-name) nor a valid [custom element name](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name) which should clearly signal that this is a special element (which will not be rendered to the DOM).
+2. Initially considered using a ':' instead of a '.' but ':' is the delimiter Aura uses to separate namespace and name.  To prevent confusion on whether this is an Aura tag, decided to use '.' instead.
+
+Additionally, `<lwc.dynamic>` is intended to be used in tandem with `lwc:is`, otherwise `<lwc:dynamic>` serves no purpose. A warning should be issued when `<lwc.dynamic>` is used without the `lwc:is` template directive.
+
+##### `lwc:is` directive
+
+A new template directive called `lwc:is` will be used to signal to compiler that the constructor is unknown at compile time. 
+
+The directive has the following properties:
+
+1. It can only be used with `<lwc.dynamic>` tag.
+    - An error will be issued when `lwc:is` is used on any other element.
+2. At compile time the value supplied to `lwc:is` can only be an expression.
+3. At runtime the expression supplied to `lwc:is` must be a `LightningElementConstructor`.
+    - An error will be thrown if anything other than a `LightningElementConstructor` is passed to `lwc:is`.
+
+#### Instantiation of the custom element
+
+Dynamic component instantiation follows the same creation path as any other LWC custom element. When the dynamic component is ready to be rendered, the element will be created and mounted to the DOM.  If there is an existing element supplied to `lwc:is`, it will be unmounted as well.
+
+This means that when the value provided to `lwc:is` changes the component and all of its children will be destroyed and recreated using the new constructor.  In addition, when the value provided to `lwc:is` is evaluated, the LWC engine will render depending on the following scenarios:
+
+- **The constructor is `undefined` or `null`**
+    - When the constructor is undefined the `<lwc.dynamic>` tag along with all of its children are not rendered to the DOM.
+- **The constructor is defined and is not a `LightningElementConstructor`**
+    - When the value provided to `lwc:is` is not a `LightningElementConstructor` then an error is thrown indicating the constructor value must be `LightningElementConstructor`.
+- **The constructor is defined and is a `LightningElementConstructor`**
+    - When the constructor is a `LightningElementConstructor`, the component will be rendered in place of `<lwc.dynamic>`. The tag name used for the dynamic component is the value of `lightningElementName` on `LightningElementConstructor`.
+
+_Note in the case of lazy loading, the component author is responsible for resolving the promise.  The only value that can be given to `lwc:is` is a `LightningElementConstructor`._
```

Comment:
Side comment: undoubtedly some folks will want to render a placeholder while the promise is resolving, but this seems to be trivial with this proposal because you can just set the constructor to the placeholder constructor while waiting for the promise to resolve.

Reply:
I agree, if they need a placeholder they can implement one and render it while the promise is resolving.  I thought about making `<lwc:dynamic>` a custom element to handle this scenario but it felt unnecessary.